### DISCUSSION TITLE
caching routes with angular universal?

### DISCUSSION THREAD

**Discussion by gabboman** [vercel/vercel#8599] [Help]
Hi!

I made a small [opensource social network](https://app.wafrn.net), and the [frontend](github.com/gabboman/wafrn) is made in angular and hosted in vercel.

I use angular universal, so vercel makes the initial load of the page, so I get SEO stuff (still working on it).

The think I would like to fix is that the initial load of posts takes some time, and that thing is not going to change too much.

What i mean is, if you load as example [this post](https://app.wafrn.net/post/e2e892ae-1b77-4a1f-8009-bbfb06670130), the first refresh and the secnd refresh will render the exact same HTML. Is there a way to tell vercel "hey cache this html until the next deployment/for a few hours"

**Comment by swarnava** [comment]
I am not super familiar with your project, but you can set cache control header using vercel.json : https://vercel.com/guides/how-to-configure-the-cache-control-response-header-in-vercel-projects#using-the-vercel.json-file

**Reply by gabboman** [reply]
can it be done through route?

also, is it browser cache or vercel cache? I am interested in the later one, as when I share a direct link to a post only the first person to load would be "the long one"
